components/application-broker/internal/nsbroker/facade.go

```diff
@@ -128,15 +121,21 @@ func (f *Facade) createServiceBroker(svcURL, namespace string) (*v1beta1.Service
 
 // Delete removes ServiceBroker and Facade. Errors don't stop execution of method. NotFound errors are ignored.
 func (f *Facade) Delete(destinationNs string) error {
-	err := f.brokerGetter.ServiceBrokers(destinationNs).Delete(NamespacedBrokerName, nil)
+	f.log.Infof("Deleting ServiceBroker from %s namespace", destinationNs)
+	err := f.brokerGetter.ServiceBrokers(destinationNs).Delete(NamespacedBrokerName, &metav1.DeleteOptions{})
 	switch {
 	case k8serrors.IsNotFound(err):
 		return nil
 	case err != nil:
 		f.log.Warnf("Deletion of namespaced-broker for namespace [%s] results in error: [%s].", destinationNs, err)
+		return err
 	}
+	err = wait.Poll(time.Second, time.Minute, func() (bool, error) {
```

Comment:
I added this check to ensure that ServiceBroker was fully removed from cluster, otherwise it could interfere with processing of the next mapping - controller would try to sync existing broker which in reality is being deleted.
At the below ss you can see the logs from execution where broker deletion took above 5s to finish: 
![image](https://user-images.githubusercontent.com/21159554/111785259-a5186280-88bc-11eb-9fbb-f075f25bc1de.png)